spatialmath/spline.py

```diff
@@ -0,0 +1,92 @@
+# Copyright (c) 2024 Boston Dynamics AI Institute LLC.
+# MIT Licence, see details in top-level file: LICENCE
+
+"""
+Classes for parameterizing a trajectory in SE3 with B-splines. 
+
+Copies parts of the API from scipy's B-spline class.
+"""
+
+from typing import Any, Optional
+from scipy.interpolate import BSpline
+from spatialmath import SE3
+import numpy as np
+import matplotlib.pyplot as plt
+from spatialmath.base.transforms3d import tranimate, trplot
+
+class BSplineSE3:
+    """ A class to parameterize a trajectory in SE3 with a 6-dimensional B-spline.
+
+    The SE3 control poses are converted to se3 twists (the lie algebra) and a B-ppline 
+    is created for each dimension of the twist, using the corresponding element of the twists 
+    as the control point for the spline.  
+
+    For detailed information about B-splines, please see this wikipedia article. 
+    https://en.wikipedia.org/wiki/Non-uniform_rational_B-spline
+    """
+
+    def __init__(self, control_poses: list[SE3], degree: int = 3, knots: Optional[list[float]] = None) -> None:
```

Comment:
```suggestion
    def __init__(self, control_poses: List[SE3], degree: int = 3, knots: Optional[List[float]] = None) -> None:
```

You need to get these from typing to support python 3.7/3.8

Reply:
> :+1: